### PR TITLE
Feature function - Batch Utils

### PR DESCRIPTION
**Pull request by GreatRSingh**
## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/utils/batch_utils.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+Utility Functions for computing features on batch.
+"""
+import numpy as np
+
+
+def coulomb_matrix_features(X_b: np.ndarray,
+                            distance_max: float = -1,
+                            distance_min: float = 18,
+                            n_distance: int = 100):
+    """Computes the values for different Feature Layers on given batch.
```
Comment (rbharath) [inline deepchem/utils/batch_utils.py:11]: Can you expand this explanation? What does the batching do? What models is it needed for? What is the scientific justification and do we have references?

File: deepchem/utils/batch_utils.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+"""
+Utility Functions for computing features on batch.
+"""
+import numpy as np
+
+
+def coulomb_matrix_features(X_b: np.ndarray,
```
Comment (rbharath) [inline deepchem/utils/batch_utils.py:7]: Let's rename this batch_coulomb_matrix_features

File: deepchem/utils/batch_utils.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                            distance_max: float = -1,
+                            distance_min: float = 18,
+                            n_distance: int = 100):
+    """Computes the values for different Feature Layers on given batch.
+    It works as a helper function to coulomb matrix.
+
```
Comment (rbharath) [inline deepchem/utils/batch_utils.py:13]: Can we add a usage example?

File: deepchem/utils/batch_utils.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
+"""
+Utility Functions for computing features on batch.
+"""
+import numpy as np
+
```
Comment (rbharath) [inline deepchem/utils/batch_utils.py:5]: Add a new section to docs/ for batch utils. Add this function and check the docs render correctly

File: deepchem/utils/batch_utils.py (hunk 1 above)
Comment (GreatRSingh) [reply deepchem/utils/batch_utils.py:11]: Done

File: deepchem/utils/batch_utils.py (hunk 2 above)
Comment (GreatRSingh) [reply deepchem/utils/batch_utils.py:7]: Done

File: deepchem/utils/batch_utils.py (hunk 3 above)
Comment (GreatRSingh) [reply deepchem/utils/batch_utils.py:13]: Done

File: deepchem/utils/batch_utils.py (hunk 4 above)
Comment (GreatRSingh) [reply deepchem/utils/batch_utils.py:5]: Done

**Review by tonydavis629 — APPROVED** [review]

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**